wasm/section.go

```diff
@@ -413,13 +419,13 @@ func (s *SectionFunctions) ReadPayload(r io.Reader) error {
 	if err != nil {
 		return err
 	}
-	s.Types = make([]uint32, count)
-	for i := range s.Types {
+	s.Types = make([]uint32, 0, getInitialCap(count))
+	for i := 0; i < int(count); i++ {
```

Comment:
```suggestion
	for i := uint32(0); i < count; i++ {
```

(on 32b architectures, `int(count)` might wrap around for big `count` values)